be/src/io/cache/fs_file_cache_storage.h

```diff
@@ -86,12 +86,19 @@ class FSFileCacheStorage : public FileCacheStorage {
 
     void load_cache_info_into_memory(BlockFileCache* _mgr) const;
 
+    using FileWriterMapKey = std::pair<UInt128Wrapper, size_t>;
+    struct FileWriterMapKeyHash {
+        std::size_t operator()(const FileWriterMapKey& w) const {
+            return UInt128Hash()(w.first.value_) + w.second;
```

Comment:
This hash function cannot guarantee uniqueness and it is not robust
1. narrow down 128bit to 64bit increase the possibility of collision 
2. the `+` result may overflow and wrap around
3. the output distribution of the hash function is not uniform

try concat 128bit input hash and the 64bit size as a byte string and hash it.
e.g.
```
char* p1 = (char*) &(w.first.value_);
char* p2 = (char*) &(w.second);
char buf[16] {p1[0:15] .. p2[0:7]};
std::string_view str(buf, 24);
return std::hash(str);
```